desc/plotting.py

```diff
@@ -2756,30 +2756,22 @@ def plot_boozer_surface(
 
     # default grids
     if grid_compute is None:
-        if eq_switch:
-            grid_kwargs = {
-                "rho": rho,
-                "M": 4 * thing.M,
-                "N": 4 * thing.N,
-                "NFP": thing.NFP,
-                "endpoint": False,
-            }
-        else:
-            grid_kwargs = {
-                "rho": rho,
-                "M": 50,
-                "N": 50,
-                "NFP": thing.NFP,
-                "endpoint": False,
-            }
+        # grid_compute only used for Equilibrium, not OmnigenousField
+        grid_kwargs = {
+            "rho": rho,
+            "M": 4 * getattr(thing, "M", 1),
+            "N": 4 * getattr(thing, "N", 1),
+            "NFP": thing.NFP,
+            "endpoint": False,
+        }
         grid_compute = _get_grid(**grid_kwargs)
     if grid_plot is None:
         grid_kwargs = {
             "rho": rho,
             "theta": 91,
             "zeta": 91,
             "NFP": thing.NFP,
-            "endpoint": True,
+            "endpoint": eq_switch,
```

Comment:
why can't there be an endpoint in the plot grid for the Omnigenous field?

Reply:
The old code used `grid_compute` for the Omnigenous field, which also had `endpoint=False` so this doesn't change that. The only thing this changes is to correctly plot on `grid_plot` instead of the `grid_compute`. 

In theory I think it should be able to use `endpoint=True`, but it was giving a worse looking plot for the test. I suspect it's an issue with `tricontour`.